src/ui-components/chart-widget/chart-widget.component.tsx

```diff
@@ -0,0 +1,81 @@
+import React from "react";
+import { Route, Link, Redirect, useHistory, useParams } from "react-router-dom";
+
+import styles from "./chart-widget.css";
+
+export default function Widget(props: any) {
+  let { patientUuid } = useParams();
+  const history = useHistory();
+
+  const hasPath = item =>
+    item.link === history.location.pathname ||
+    item.link === props.paths.medications;
+
+  function getInitialSelected() {
+    const i = props.widgetConfig.routes.findIndex(hasPath);
+    return i === -1 ? 0 : i;
+  }
+
+  const [selected, setSelected] = React.useState(getInitialSelected());
+
+  function handleClick(index, path) {
+    setSelected(index);
+    props.setLastRoute(path);
+  }
+
+  return (
+    <>
+      <nav className={styles.summariesnav} style={{ marginTop: "0" }}>
+        <ul>
+          {props.widgetConfig.routes.map((item, index) => {
+            return (
+              <li key={index}>
+                <div
+                  className={`${
+                    index === selected ? styles.selected : styles.unselected
+                  }`}
+                >
+                  <Link to={item.link}>
+                    <button
+                      className="omrs-unstyled"
+                      onClick={() => handleClick(index, item.link)}
+                    >
+                      {item.name}
+                    </button>
+                  </Link>
+                </div>
+              </li>
+            );
+          })}
+        </ul>
+      </nav>
+      <Route
+        exact
+        path={`/patient/${patientUuid}/chart/${props.widgetConfig.name}`}
+      >
+        {props.paths[props.widgetConfig.name] === "" ||
+        props.paths[props.widgetConfig.name] ===
+          `/patient/${patientUuid}/chart/${props.widgetConfig.name}` ? (
+          <Redirect
+            to={`/patient/${patientUuid}/chart/${props.widgetConfig.name}/${props.widgetConfig.defaultRoute}`}
+          />
+        ) : (
+          <Redirect to={props.paths[props.widgetConfig.name]} />
+        )}
+      </Route>
+
+      {props.widgetConfig.routes.map((item, index) => {
```

Comment:
@fatmali , I'm struggling with where exactly routes should be declared. This component takes a config (see medications.component.tsx  below for en example) and: 1) creates a component with a nav bar and 2) declares the routes for that component. 

Should the routes be declared here? Should we be centrally declaring all routes in one file? The latter seems like a bit unwieldy but at least easy to figure out what routes are being declared. The former seems a little more logical, it's easier to associate a feature with routes. 

Interested to hear your thoughts.

Reply:
Having all the routes in one file is going to be difficult to manage and to maintain. I think it would be much better to split it up into different files.